pkg/query/conv.go

```diff
@@ -1,10 +1,56 @@
 package query
 
 import (
+	"strconv"
+
 	sitter "github.com/smacker/go-tree-sitter"
 	"go.lsp.dev/protocol"
 )
 
+func Unquote(input []byte, n *sitter.Node) string {
+done:
+	for {
+		switch n.Type() {
+		case NodeTypeModule,
+			NodeTypeBlock,
+			NodeTypeExpressionStatement:
+			n = n.Child(0)
+		case NodeTypeString:
+			break done
+		default:
+			return ""
+		}
+	}
+
+	startDelim := n.Child(0)
+	endDelim := n.Child(int(n.ChildCount() - 1))
+	byteoffset := startDelim.EndByte()
+	bytes := []byte{}
+
+	for i := 1; i < int(n.ChildCount()-1); i++ {
+		escape := n.Child(i)
```

Comment:
Can you add a comment explaining `n`'s children? I'm struggling to read this even with consulting the treesitter grammar. Or maybe this is illustratable with a treesitter playground link?

Presumably the first and last children are the quotes themselves. This code is referring to every child in between as an `escape`, which leaves me confused where the `foo` in `"foo"` would be handled, since that doesn't have an escape. Do we have an iteration where escape == "foo" (or maybe one iteration per char)?

or I guess unescaped chars are handled by the next if statement because they don't belong to a node at all?

Reply:
Yeah, there are no nodes representing the unescaped characters in the string. Let me see if I can explain that for posterity.

Reply:
9cde4ca79b810596815d90eb582a52e08b9ffffe

Reply:
9cde4ca79b810596815d90eb582a52e08b9ffffe